lib/_emerge/depgraph.py

```diff
@@ -7622,6 +7622,10 @@ def _wrapped_select_pkg_highest_available_imp(
                     if pkg in self._dynamic_config._runtime_pkg_mask:
                         # The package has been masked by the backtracking logic
                         continue
+                    if empty and pkg.installed:
```

Comment:
Could you see if you could construct a testcase to add?

Reply:
Should test to verify that --exclude allows an installed package.

Reply:
Thanks @zmedico , I made a small adjustment. I tested manually whenever `--exclude` is there, the package is not reinstalled. With my original modification emerge just fails, because the excluded package is masked.

Reply:
I will add tests soon.

Reply:
Tests are done. Could you rereview?